astropy/wcs/wcs.py

```diff
@@ -389,6 +389,9 @@ def __init__(self, header=None, fobj=None, key=' ', minerr=0.0,
                     header_string = header
             elif isinstance(header, fits.Header):
                 header_string = header.tostring().rstrip()
+            elif type(header) == fits.ImageHDU or type(header) == fits.PrimaryHDU:
+                header = header.header
+                header_string = header.tostring().rstrip()
```

Comment:
It occurs to me that this line is repeated at the end of every branch in this conditional.  It should just be moved after the if/else. The last clause in the conditional does create a new variable `new_header`, but it could just as easily write
```
orig_header = header
header = fits.Header()
```